### PR TITLE
feat: v3 unbalanced removes

### PR DESCRIPTION
**Pull request by agualis**
Adds v3 removes using standard permit signatures 
Works for `Proportional` and `Single token` removes

Example of Proportional remove (using sepolia pool in test api):

permitRemoves

### CODE REVIEW

File: lib/modules/transactions/transaction-steps/useSignPermitStep.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          isLoading={isLoading}
+          onClick={signPermit}
+          loadingText={buttonLabel}
+        >
+          <HStack spacing="sm" width="100%">
+            <Text fontWeight="bold" color={primaryTextColor}>
+              {buttonLabel}
+            </Text>
+            <Spacer />
+            <SignIcon size={16} />
```
Comment (agualis) [inline lib/modules/transactions/transaction-steps/useSignPermitStep.tsx:50]: @uiuxxx @garethfuller 
I added SignIcon as an example here but I prefer to create a separate PR where we review all the labels and custom icons.

File: lib/modules/tokens/approvals/permit/useSignPermit.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    error,
+  }
+}
+
+function getButtonLabel(signPermitState: SignatureState, poolSymbol?: string) {
+  if (signPermitState === SignatureState.Ready) return getReadyLabel(poolSymbol)
+  if (signPermitState === SignatureState.Confirming) return 'Confirm permit signature in wallet'
+  if (signPermitState === SignatureState.Preparing) return 'Preparing'
+  if (signPermitState === SignatureState.Completed) return 'Permit Signed'
+  return ''
```
Comment (garethfuller) [inline lib/modules/tokens/approvals/permit/useSignPermit.tsx:99]: Again I think this should end up being different wording.
```suggestion
  if (signPermitState === SignatureState.Confirming) return 'Confirm approval with signature'
  if (signPermitState === SignatureState.Preparing) return 'Preparing'
  if (signPermitState === SignatureState.Completed) return 'Approval signed'
  return ''
```

File: lib/shared/labels.ts [hunk 3]
```diff
@@ -1,4 +1,10 @@
...
+// Common labels used in app, helps keep common text consistent and easy to change.
 export const LABELS = {
   walletNotConnected: 'Wallet not connected',
 }
+
+// Converts an index to a letter. 0 is A, 1 is B, etc.
+export function indexToLetter(index: number): string {
+  if (index >= 0 && index <= 25) return String.fromCharCode(index + 65)
+  throw new Error('index but be between 0 and 25')
+}
```
Comment (garethfuller) [inline lib/shared/labels.ts:10]: Why is this function needed?

Comment (agualis) [reply lib/shared/labels.ts:10]: Is is used in the Step Tracker to show A, B, C in the sub steps here:

https://github.com/balancer/frontend-v3/blob/bcee8e484d88a99813c22a9ed656a2712e0b7371/lib/modules/transactions/transaction-steps/step-tracker/Step.tsx#L85

Screenshot 2024-10-08 at 18 07 08

**Pull request merged**